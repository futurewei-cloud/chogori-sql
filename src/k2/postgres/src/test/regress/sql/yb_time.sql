--
-- TIME
--

CREATE TABLE TIME_TBL (f1 time(2));

INSERT INTO TIME_TBL VALUES ('00:00');
INSERT INTO TIME_TBL VALUES ('01:00');
-- as of 7.4, timezone spec should be accepted and ignored
INSERT INTO TIME_TBL VALUES ('02:03 PST');
INSERT INTO TIME_TBL VALUES ('11:59 EDT');
INSERT INTO TIME_TBL VALUES ('12:00');
INSERT INTO TIME_TBL VALUES ('12:01');
INSERT INTO TIME_TBL VALUES ('23:59');
INSERT INTO TIME_TBL VALUES ('11:59:59.99 PM');
INSERT INTO TIME_TBL VALUES ('24:00');
INSERT INTO TIME_TBL VALUES ('23:59:59.999');

INSERT INTO TIME_TBL VALUES ('2003-03-07 15:36:39 America/New_York');
INSERT INTO TIME_TBL VALUES ('2003-07-07 15:36:39 America/New_York');
-- this should fail (the timezone offset is not known)
INSERT INTO TIME_TBL VALUES ('15:36:39 America/New_York');

-- Out of range values
INSERT INTO TIME_TBL VALUES ('25:00');
INSERT INTO TIME_TBL VALUES ('24:01');

SELECT f1 AS "Time" FROM TIME_TBL ORDER BY f1;

SELECT f1 AS "Three" FROM TIME_TBL WHERE f1 < '05:06:07' ORDER BY f1;

SELECT f1 AS "Five" FROM TIME_TBL WHERE f1 > '05:06:07' ORDER BY f1;

SELECT f1 AS "None" FROM TIME_TBL WHERE f1 < '00:00' ORDER BY f1;

SELECT f1 AS "Eight" FROM TIME_TBL WHERE f1 >= '00:00' ORDER BY f1;

-- ASC/DESC check
SELECT * FROM TIME_TBL ORDER BY f1 ASC;
SELECT * FROM TIME_TBL ORDER BY f1 DESC;

--
-- TIME simple math
--
-- We now make a distinction between time and intervals,
-- and adding two times together makes no sense at all.
-- Leave in one query to show that it is rejected,
-- and do the rest of the testing in horology.sql
-- where we do mixed-type arithmetic. - thomas 2000-12-02

SELECT f1 + time '00:01' AS "Illegal" FROM TIME_TBL;

--
-- Test for PRIMARY KEY
--

CREATE TABLE time_data_with_pk(id TIME PRIMARY KEY, val time);
INSERT INTO time_data_with_pk VALUES ('01:01:01','01:11:21');
INSERT INTO time_data_with_pk VALUES ('02:02:02','02:12:22');
INSERT INTO time_data_with_pk VALUES ('03:03:03','03:13:23');
SELECT * FROM time_data_with_pk ORDER BY id;
SELECT VAL FROM time_data_with_pk WHERE id = '02:02:02';
