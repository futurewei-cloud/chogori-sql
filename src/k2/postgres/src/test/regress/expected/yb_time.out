--
-- TIME
--
CREATE TABLE TIME_TBL (f1 time(2));
INSERT INTO TIME_TBL VALUES ('00:00');
INSERT INTO TIME_TBL VALUES ('01:00');
-- as of 7.4, timezone spec should be accepted and ignored
INSERT INTO TIME_TBL VALUES ('02:03 PST');
INSERT INTO TIME_TBL VALUES ('11:59 EDT');
INSERT INTO TIME_TBL VALUES ('12:00');
INSERT INTO TIME_TBL VALUES ('12:01');
INSERT INTO TIME_TBL VALUES ('23:59');
INSERT INTO TIME_TBL VALUES ('11:59:59.99 PM');
INSERT INTO TIME_TBL VALUES ('24:00');
INSERT INTO TIME_TBL VALUES ('23:59:59.999');
INSERT INTO TIME_TBL VALUES ('2003-03-07 15:36:39 America/New_York');
INSERT INTO TIME_TBL VALUES ('2003-07-07 15:36:39 America/New_York');
-- this should fail (the timezone offset is not known)
INSERT INTO TIME_TBL VALUES ('15:36:39 America/New_York');
ERROR:  invalid input syntax for type time: "15:36:39 America/New_York"
LINE 1: INSERT INTO TIME_TBL VALUES ('15:36:39 America/New_York');
                                     ^
-- Out of range values
INSERT INTO TIME_TBL VALUES ('25:00');
ERROR:  date/time field value out of range: "25:00"
LINE 1: INSERT INTO TIME_TBL VALUES ('25:00');
                                     ^
INSERT INTO TIME_TBL VALUES ('24:01');
ERROR:  date/time field value out of range: "24:01"
LINE 1: INSERT INTO TIME_TBL VALUES ('24:01');
                                     ^
SELECT f1 AS "Time" FROM TIME_TBL ORDER BY f1;
    Time     
-------------
 00:00:00
 01:00:00
 02:03:00
 11:59:00
 12:00:00
 12:01:00
 15:36:39
 15:36:39
 23:59:00
 23:59:59.99
 24:00:00
 24:00:00
(12 rows)

SELECT f1 AS "Three" FROM TIME_TBL WHERE f1 < '05:06:07' ORDER BY f1;
  Three   
----------
 00:00:00
 01:00:00
 02:03:00
(3 rows)

SELECT f1 AS "Five" FROM TIME_TBL WHERE f1 > '05:06:07' ORDER BY f1;
    Five     
-------------
 11:59:00
 12:00:00
 12:01:00
 15:36:39
 15:36:39
 23:59:00
 23:59:59.99
 24:00:00
 24:00:00
(9 rows)

SELECT f1 AS "None" FROM TIME_TBL WHERE f1 < '00:00' ORDER BY f1;
 None 
------
(0 rows)

SELECT f1 AS "Eight" FROM TIME_TBL WHERE f1 >= '00:00' ORDER BY f1;
    Eight    
-------------
 00:00:00
 01:00:00
 02:03:00
 11:59:00
 12:00:00
 12:01:00
 15:36:39
 15:36:39
 23:59:00
 23:59:59.99
 24:00:00
 24:00:00
(12 rows)

-- ASC/DESC check
SELECT * FROM TIME_TBL ORDER BY f1 ASC;
     f1
-------------
 00:00:00
 01:00:00
 02:03:00
 11:59:00
 12:00:00
 12:01:00
 15:36:39
 15:36:39
 23:59:00
 23:59:59.99
 24:00:00
 24:00:00
(12 rows)

SELECT * FROM TIME_TBL ORDER BY f1 DESC;
     f1
-------------
 24:00:00
 24:00:00
 23:59:59.99
 23:59:00
 15:36:39
 15:36:39
 12:01:00
 12:00:00
 11:59:00
 02:03:00
 01:00:00
 00:00:00
(12 rows)

--
-- TIME simple math
--
-- We now make a distinction between time and intervals,
-- and adding two times together makes no sense at all.
-- Leave in one query to show that it is rejected,
-- and do the rest of the testing in horology.sql
-- where we do mixed-type arithmetic. - thomas 2000-12-02
SELECT f1 + time '00:01' AS "Illegal" FROM TIME_TBL;
ERROR:  operator is not unique: time without time zone + time without time zone
LINE 1: SELECT f1 + time '00:01' AS "Illegal" FROM TIME_TBL;
                  ^
HINT:  Could not choose a best candidate operator. You might need to add explicit type casts.
--
-- Test for PRIMARY KEY
--
CREATE TABLE time_data_with_pk(id TIME PRIMARY KEY, val time);
INSERT INTO time_data_with_pk VALUES ('01:01:01','01:11:21');
INSERT INTO time_data_with_pk VALUES ('02:02:02','02:12:22');
INSERT INTO time_data_with_pk VALUES ('03:03:03','03:13:23');
SELECT * FROM time_data_with_pk ORDER BY id;
    id    |   val
----------+----------
 01:01:01 | 01:11:21
 02:02:02 | 02:12:22
 03:03:03 | 03:13:23
(3 rows)

SELECT VAL FROM time_data_with_pk WHERE id = '02:02:02';
   val
----------
 02:12:22
(1 row)

